client/state/ui/editor/actions.js

```diff
@@ -225,6 +233,7 @@ export function editorReset( options ) {
 	return {
 		type: EDITOR_RESET,
 		isLoading: get( options, 'isLoading', false ),
+		isLoaded: get( options, 'isLoaded', false ),
```

Comment:
You probably don't need this `isLoaded` option, as it's never set to a non-default value like `true`. `EDITOR_RESET` always sets `isLoaded` to `false`.

Similarly, `EDITOR_LOADED` always sets `isLoaded` to `true`. Therefore, the `EDITOR_LOADED` action object doesn't need the `isLoaded` flag.

The reducer is then very simple:
```js
function isLoaded( state = false, action ) {
  switch ( action.type ) {
    case EDITOR_LOADED:
      return true;
    case EDITOR_RESET:
      return false;
    default:
      return state;
  }
}
```

Reply:
Thanks for the pointers, @jsnajdr. I just tried refactoring per your suggestions, however I think I'm missing something as the following doesn't quite work 100%:

1. Removed `isLoaded` flag, so `editorLoaded` method simply reads:

```
export const editorLoaded = () => ( {
	type: EDITOR_LOADED,
} )
```

2. Remove `isLoaded` from `editorReset` action:

```
export function editorReset( options ) {
	return {
		type: EDITOR_RESET,
		isLoading: get( options, 'isLoading', false ),
		loadingError: get( options, 'loadingError', null ),
	};
}
```

3. Refactor reducer per the suggestion above (i.e. switch on both `EDITOR_LOADED` and `EDITOR_RESET`).

This works, except the state of `isLoaded` is never toggled back to `false` after the reset occurs. I think I'm missing something really basic here - how do I "unset" the `EDITOR_LOADED` action in the `editorReset`, now that it doesn't have an associated flag?

Reply:
The `EDITOR_RESET` `isLoaded` option can be reverted now.

Reply:
Reverted `EDITOR_RESET` changes in 42f4f01